### PR TITLE
Add Reducers & actions

### PR DESCRIPTION
**Pull request by Sirvelasque**
# Add reducers
- book reducer
- categories reducer
- ConfigureStore
# Add actions
- Add books
- Remove book

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { configureStore } from 'redux';
```
Comment (Gambit142) [inline src/redux/configureStore.js]: - [x] Kindly import configureStore from `@reduxjs/toolkit` and not `redux`. 👍

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import booksReducer from './books/books';
+import categoriesReducers from './categories/categories';
+
+const store = configureStore({
+  rootReducer: {
+    booksReducer,
+    categoriesReducers,
+  },
+});
```
Comment (Gambit142) [inline src/redux/configureStore.js]: - [x] [Required] Configure store accepts a single configuration object that contains a `reducer` key and some other keys. To configure your store, you need to use the `reducer` key in place of the `rootReducer` key.

```javascript
const store = configureStore({
  reducer: {
   books: booksReducer,
   categories: statusReducer
  },
});
```

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  switch (action.type) {
+    case ADD:
+      return [
+        ...state,
+        action.book,
+      ];
+
+    case REMOVE:
+      return state.filter((book, index) => index !== action.index);
+
```
Comment (Gambit142) [inline src/redux/books/books.js]: - [x] When you use the second parameter in the filter method called `index`, you are referring to the index (number) position of the item in the array which starts from 0. Unless you want to set the ids of each item to the index, I would advise you to not implement the case for removing a book this way. You can use the id of the book to make your comparison.

#### Hint:  `return state.filter(book => book.id !== action.index);`

File: package.json [hunk 4]
```diff
@@ -9,8 +9,10 @@
     "prop-types": "^15.8.1",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
+    "react-redux": "^8.0.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
+    "redux": "^4.2.0",
     "uuid": "^8.3.2",
     "uuidv4": "^6.2.13"
   },
```
Comment (Gambit142) [inline package.json:19]: - [x] Kindly install the `redux toolkit` as part of your project dependencies. `npm install @reduxjs/toolkit`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Sirvelasque 👋, 

Good job so far!

## Highlights
- All linters checks passed 💯 
- Profesional `README.md` file 💯 
- File structure of the project follows the `duck pattern` 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

#### You can also consider using the dash (-) or underscore (_) as part of a branch name when creating a branch. Please do not use the character `&` as part of the branch name to create a branch. For example `Reducers-Actions` or `Reducers_Actions` and not `Reducers&Actions`. When you use the character `&` and I try to switch to the branch, I get errors in the terminal. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Sirvelasque 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done addressing the required changes from the previous reviewer.
- All seems good

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**